theano/compile/tests/test_nanguardmode.py

```diff
@@ -58,7 +58,7 @@ def test_NanGuardMode():
         numpy.asarray(1e20).astype(theano.config.floatX), (3, 4, 5))
 
     x = T.tensor3()
-    y = x[:, T.arange(2), T.arange(2)]
+    y = x[:, T.arange(2), T.arange(2), None]
```

Comment:
This cause an error in the test::

======================================================================

ERROR: Tests if NanGuardMode is working by feeding in numpy.inf and numpy.nans

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/miniconda2/envs/pyenv/lib/python2.6/site-packages/nose/case.py", line 197, in runTest

    self.test(*self.arg)

  File "/home/travis/build/Theano/Theano/theano/compile/tests/test_nanguardmode.py", line 61, in test_NanGuardMode

    y = x[:, T.arange(2), T.arange(2), None]

  File "/home/travis/build/Theano/Theano/theano/tensor/var.py", line 506, in __getitem__

    return theano.tensor.subtensor.advanced_subtensor(self, *args)

  File "/home/travis/build/Theano/Theano/theano/gof/op.py", line 602, in __call__

    node = self.make_node(*inputs, **kwargs)

  File "/home/travis/build/Theano/Theano/theano/tensor/subtensor.py", line 2135, in make_node

    bcast = adv_index_broadcastable_pattern(x, index)

  File "/home/travis/build/Theano/Theano/theano/tensor/subtensor.py", line 2116, in adv_index_broadcastable_pattern

    retshape = numpy.empty(fakeshape)[newidx].shape

TypeError: long() argument must be a string or a number, not 'NoneType'

----------------------------------------------------------------------

Ran 993 tests in 1863.014s



But I think you want something else. Here the None will trigger a new axis. It won't cause a None in the graph.

What you want to test is the graph line in 
```
import theano
from theano.tensor.type_other import make_slice

    # In the past, this was crahsing during compilation.
    i = theano.tensor.iscalar()
    s1 = make_slice(0, i)
    f = theano.function([i], [s1], ...
```